analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,214 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path.
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm).
+    The Impact parameter significance is stored in the particle's path.
+    
+    New attributes for particles.path (from compute_IP) :
+    *   closest_t = time of closest approach to the primary vertex.
+    *   IP = signed impact parameter
+    *   IPx, IPy, IPz = coordinates of the point of closest approach to the
+        primary vertex, with respets to the primary vertex
+        
+    New attributes for particles.path (from compute_theta_0) :
```

Comment:
Documentation much improved!

I would still like that we encapsulate all b tagging information into some container class, say BTagInfo, for both particles and jets. Maybe we need to class. 
The reason for that is that it is dangerous to add attributes to an object after creation (can crush existing attribute, user has no obvious way to know the attributes are there by looking at the class code). So better to put all new information related to b tagging into a single new attribute.